src/main/java/Duke.java

```diff
@@ -1,10 +1,55 @@
+/**
+ * The driver class for the Duke program.
+ * Reads user input and adds, deletes, updates and saves the user's tasks accordingly.
+ */
 public class Duke {
+    private Ui ui;
+    private Storage store;
+    private TaskList taskList;
+    private Parser parser;
+
+    /**
+     * Constructor for Duke that takes in a String representing the
+     * name of the save file for the user's task list.
+     * @param fileName String of the save file of the user's task list.
+     */
+    public Duke(String fileName) {
+        // Initialize Ui
+        this.ui = new Ui();
+        // Initialize storage
+        this.store = new Storage(fileName);
+        // Initialize taskList
+        this.taskList =  new TaskList(this.store.loadFromFilePath());
+        this.parser = new Parser();
+    }
+
+    /**
+     * Driver function that runs the Duke program.
+     */
+    public void run() {
```

Comment:
Really nice and clean run() function man. I love the separation of Command class to call execute.